### ISSUE TITLE
improve dependency error message output

### ISSUE THREAD

**Issue by rmarquis** [rmarquis/pacaur#173]
```
$ pacaur -S xmp 
:: Package(s) xmp not found in repositories, trying AUR...
:: resolving dependencies...
:: no results found for bmp
```

bmp is a makedep of the dep libxmp. A more explicit error message, such as "no result found for dependency bmp" would be very welcome.

For reference, here is a similar pacman output:

```
loading packages...
resolving dependencies...
warning: cannot resolve "enemy-territory-data", a dependency of "etlegacy-mod-git"
warning: cannot resolve "etlegacy-mod-git", a dependency of "etlegacy-git"
warning: cannot resolve "enemy-territory-data", a dependency of "etlegacy-mod-git"
:: The following packages cannot be upgraded due to unresolvable dependencies:
      etlegacy-git  etlegacy-mod-git

:: Do you want to skip the above packages for this upgrade? [y/N] 
```

**Comment by brainpower** [comment]
I don't know if this is of the same origin the issue mentiond above, so I'll comment here first before opening a new issue.

Today I wanted to do a full update, on some package to be updated, was a new dependency added, attica-git, it conflicts with attica-qt5.
I didn't want to use the -git version so I simply wanted to skip the package which added that dependency.
It would be nice if pacaur could tell me somehow which package wanted that dependency.
Even if its only after I say no to "Remove attica-qt5? [y/N]".
I read above that

> There is currently no separate tracking of which dependencies are required by a certain installed package [...]

Does that apply in my situation, too?

    % LC_ALL=C pacaur -Sua
    :: Starting AUR upgrade...
    :: resolving dependencies...
    :: looking for inter-conflicts...
    :: attica-git and attica-qt5 are in conflict (attica-qt5). Remove attica-qt5? [y/N] n
    :: unresolvable package conflicts detected
    :: failed to prepare transaction (conflicting dependencies)
    :: attica-git and attica-qt5 are in conflict

**Comment by rmarquis** [comment]
@brainpower > Yes, this is exactly what this issue is about. You'll need to find which dependency needs attica-git and use the --ignore flag with it. This is a really inconvenient, I agree.

**Comment by flying-sheep** [comment]
now that we have .SRCINFO, constructing the dependency tree should be much easier.

**Comment by rmarquis** [comment]
Not really, but since pacaur recently moved to tsort for dependency resolution (#408) I hope I'll find a nice way to achieve this much needed improvement.

**Comment by rmarquis** [comment]
Initial error tracker implemented in ad7813eee9e6d6c7d8f3b5522fb74e053388937b. Tsort definitely helped a great deal here!

The output looks like this:

```
$ pacaur -S plasma-desktop-git
:: Package plasma-desktop-git not found in repositories, trying AUR...
:: resolving dependencies...
:: no results found for prison-frameworks (dependency tree: plasma-desktop-git kmenuedit-git khotkeys-git plasma-workspace-git prison-frameworks)
```

```
pacaur -S aura                                                                                                                                                                                        
:: Package aura not found in repositories, trying AUR...
:: resolving dependencies...
:: no results found for haskell-regex-pcre-builtin (dependency tree: aura haskell-regex-pcre-builtin)
:: no results found for haskell-servant (dependency tree: aura haskell-aur haskell-servant)
:: no results found for haskell-servant-client (dependency tree: aura haskell-aur haskell-servant-client)
```

This needs some cleaning, and is implemented at the moment for missing dependency only. The `--ignore` case discussed above still needs to be taken into account.

Feedback welcome!

**Comment by Deewiant** [comment]
Looks great! The only downgrade I can see is that version bounds are no longer emitted. Using your `aura` example, pacaur 4.6.10 says:

```
$ pacaur -Sa aura
:: resolving dependencies...
:: no results found for haskell-regex-pcre-builtin
:: no results found for haskell-aeson<1
:: no results found for haskell-servant-client>=0.9
:: no results found for haskell-servant>=0.9
```

Note the `<1`, `>=0.9` output. pacaur 4.6.10.91.gad7813e only reports the package names, like you already quoted — and apparently it doesn't report `haskell-aeson` at all, perhaps because the package itself can be found, just with an unacceptable version number? (I wonder what would happen if that were the only missing dependency?) I think what you have done is definitely an improvement anyway but it'd be even better if it could output the version bounds as well. (Bonus points for reporting "package not found" and "package found but not with right version" differently!)

**Comment by rmarquis** [comment]
@Deewiant Good catch, I'll have a look shortly.

**Comment by rmarquis** [comment]
Fixed in 912ef4cbe5d3b330fc67d363b8dddf8fd03fffd2:

```
$ pacaur -S aura
:: Package aura not found in repositories, trying AUR...
:: resolving dependencies...
:: no results found for haskell-regex-pcre-builtin (dependency tree: aura haskell-regex-pcre-builtin)
:: no results found for haskell-aeson<1 (dependency tree: aura haskell-aur haskell-aeson)
:: no results found for haskell-servant-client>=0.9 (dependency tree: aura haskell-aur haskell-servant-client)
:: no results found for haskell-servant>=0.9 (dependency tree: aura haskell-aur haskell-servant)
```

I do not think a different error message is necessary since all that info is contained in the version number already.

**Comment by gandalf3** [comment]
I'm getting a strange behavior which seems related to this. 
Attempting an aur upgrade gives me the following:
```
:: no results found for python2-qscintilla-qt4<2.10 (dependency tree: virtualbox-ext-oracle python2-qscintilla-qt4)
:: no results found for python2-sip<4.19.1 (dependency tree: virtualbox-ext-oracle python2-sip)
:: no results found for python-qscintilla-qt4-common<2.10 (dependency tree: virtualbox-ext-oracle python-qscintilla-qt4-common)
:: no results found for qscintilla-qt4<2.10 (dependency tree: virtualbox-ext-oracle qscintilla-qt4)
:: no results found for sip<4.19.1 (dependency tree: virtualbox-ext-oracle sip)
```
There are several things odd about this:
* All the packages and versions which couldn't be found are in the official repos and have satisfactory versions installed already.
* The dependent package (in this case virtualbox-ext-oracle) has seemingly no relation to the packages listed as missing.
* If I manually update the dependent package and run pacaur again, I'll get the same error with another unrelated, but out-of-date aur package. It appears to be going through all outdated aur packages and claiming these five packages from extra are missing dependencies for each of them.

**Comment by rmarquis** [comment]
@gandalf3 Could you open a new issue with the additional debug output? Thx.
